monitoring/generator.go

```diff
@@ -694,7 +694,7 @@ func (c *Container) promAlertsFile() *promRulesFile {
 							fireOnNan = "0"
 						}
 						alertQuery = fmt.Sprintf("((%s) >= 0) OR on() vector(%v)", alertQuery, fireOnNan)
-						group.AppendRow(alertQuery, makeLabels("low"))
+						group.AppendRow(fmt.Sprintf("min(%s)", alertQuery), makeLabels("low"))
```

Comment:
Important to document _why_ this is here

```suggestion
						// Wrap the query in min() so that if there are multiple series (e.g. per-container) they
						// get flattened into a single one (we only support per-service alerts,
						// not per-container/replica).
						// More context: https://github.com/sourcegraph/sourcegraph/issues/11571#issuecomment-654571953
						group.AppendRow(fmt.Sprintf("min(%s)", alertQuery), makeLabels("low"))
```